cluster-service/deploy/templates/k8s-job-arbitrary-commands.yaml

```diff
@@ -0,0 +1,103 @@
+{{ if .Values.deployDebugJob }}
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: cs-arbitrary-commands
+  namespace: '{{ .Release.Namespace }}'
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: cs-arbitrary-commands-role
+  namespace: '{{ .Release.Namespace }}'
+rules:
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apps"]
+    resources: ["deployments"]
+    verbs: ["get", "list", "watch"]
+---
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: cs-arbitrary-commands-rolebinding
+  namespace: '{{ .Release.Namespace }}'
+subjects:
+  - kind: ServiceAccount
+    name: cs-arbitrary-commands
+    namespace: '{{ .Release.Namespace }}'
+roleRef:
+  kind: Role
+  name: cs-arbitrary-commands-role
+  apiGroup: rbac.authorization.k8s.io
+---
+apiVersion: batch/v1
+kind: Job
+metadata:
+  name: cs-arbitrary-commands
```

Comment:
The K8s Job name is 'cs-arbitrary-commands'. If we deploy two times, what is going to happen? I understand it would fail because of the resource already existing.

At the same time I think it might work because there's a ttlSecondsAfterFinished of 5 minutes and it is unlikely two rollouts will occur with more frequency than that.

Still, I wonder if having a unique name each time is deployed would be better to avoid any possibility of collision?

Reply:
Yes, ttlSecondsAfterFinished should handle deleting the job before the next one runs. However, as suggested, I will also add some random characters to the job name to ensure each job execution has a unique name.